### PR TITLE
Fix some psalm findings

### PR DESCRIPTION
**Pull request by franmomu**
|      Q       |   A
|------------- | -----------
| Type         | improvement
| BC Break     | yes/no
| Fixed issues | 

#### Summary

Some other fixes for https://github.com/doctrine/mongodb-odm/pull/2329

### CODE REVIEW

File: lib/Doctrine/ODM/MongoDB/Mapping/ClassMetadata.php [hunk 1]
```diff
@@ -65,7 +65,7 @@
  *      association?: int,
  *      id?: bool,
  *      collectionClass?: class-string,
- *      cascade?: list<string>|string,
+ *      cascade?: array<string>|string,
```
Comment (franmomu) [inline lib/Doctrine/ODM/MongoDB/Mapping/ClassMetadata.php]: this change comes from

https://github.com/doctrine/mongodb-odm/blob/5aa06c040833c2de4dcb2cc5bd023265db3b8d9c/tests/Doctrine/ODM/MongoDB/Tests/Mapping/AbstractMappingDriverTest.php#L710

where `cascade` also admits integer offsets

File: lib/Doctrine/ODM/MongoDB/Tools/Console/Command/Schema/CreateCommand.php [hunk 2]
```diff
@@ -39,8 +39,8 @@ protected function configure()
 
     protected function execute(InputInterface $input, OutputInterface $output)
     {
-        $create = array_filter($this->createOrder, static function ($option) use ($input) {
-            return $input->getOption($option);
+        $create = array_filter($this->createOrder, static function (string $option) use ($input): bool {
+            return (bool) $input->getOption($option);
```
Comment (franmomu) [inline lib/Doctrine/ODM/MongoDB/Tools/Console/Command/Schema/CreateCommand.php:43]: https://github.com/doctrine/mongodb-odm/blob/5aa06c040833c2de4dcb2cc5bd023265db3b8d9c/lib/Doctrine/ODM/MongoDB/Tools/Console/Command/Schema/CreateCommand.php#L25

`$this->createOrder` are strings, and `array_filter` expects to return a scalar value.

The same applies to `DropCommand`.

File: tests/Doctrine/ODM/MongoDB/Tests/Functional/LockTest.php [hunk 3]
```diff
@@ -570,6 +570,8 @@ abstract class AbstractVersionBase
     /** @ODM\EmbedMany(targetDocument=Issue::class) */
     public $issues;
 
+    public $version;
+
```
Comment (franmomu) [inline tests/Doctrine/ODM/MongoDB/Tests/Functional/LockTest.php:574]: It has a `getVersion`, but there was no attribute:

https://github.com/doctrine/mongodb-odm/blob/5aa06c040833c2de4dcb2cc5bd023265db3b8d9c/tests/Doctrine/ODM/MongoDB/Tests/Functional/LockTest.php#L589-L592

File: tests/Doctrine/ODM/MongoDB/Tests/Functional/Ticket/GH453Test.php [hunk 4]
```diff
@@ -146,7 +146,7 @@ public function testEmbedMany()
         $this->assertBsonArray($doc->id, 'embedManyAddToSet');
 
         // Check that the value is changed properly
-        unset($colPush[1], $colSet[1], $colSetArray[1], $colAddToSet['1']);
+        unset($colPush[1], $colSet[1], $colSetArray[1], $colAddToSet[1]);
```
Comment (franmomu) [inline tests/Doctrine/ODM/MongoDB/Tests/Functional/Ticket/GH453Test.php:149]: I think that was a mistake, AFAIK being a numeric string it is the same result as `1`.

File: tests/Doctrine/ODM/MongoDB/Tests/Mapping/ClassMetadataTest.php [hunk 5]
```diff
@@ -418,8 +418,7 @@ public function testSetFieldValueWithProxy()
 
     public function testSetCustomRepositoryClass()
     {
-        $cm            = new ClassMetadata(self::class);
-        $cm->namespace = 'Doctrine\ODM\MongoDB\Tests\Mapping';
```
Comment (franmomu) [inline tests/Doctrine/ODM/MongoDB/Tests/Mapping/ClassMetadataTest.php:422]: There is no `$namespace` attribute in [`ClassMetadata`](https://github.com/doctrine/mongodb-odm/blob/2.3.x/lib/Doctrine/ODM/MongoDB/Mapping/ClassMetadata.php).

File: lib/Doctrine/ODM/MongoDB/Mapping/ClassMetadata.php (hunk 1 above)
Comment (alcaeus) [reply lib/Doctrine/ODM/MongoDB/Mapping/ClassMetadata.php]: I think we can leave this as `array` and remove the integer keys where used. This was indeed expected to be a simple list.

File: tests/Doctrine/ODM/MongoDB/Tests/Functional/Ticket/GH453Test.php (hunk 4 above)
Comment (alcaeus) [reply tests/Doctrine/ODM/MongoDB/Tests/Functional/Ticket/GH453Test.php:149]: I agree.

**Review by alcaeus — APPROVED** [review]

**Comment by malarzm** [comment]
Thanks @franmomu!

**Pull request merged**